credsweeper/deep_scanner/deep_scanner.py

```diff
@@ -191,8 +191,8 @@ def structure_scan(
                                                             file_path=struct_provider.file_path,
                                                             file_type=struct_provider.file_type,
                                                             info=f"{struct_provider.info}|STRUCT:{key}")
-                candidates.extend(self.structure_scan(val_struct_provider, depth, recursive_limit_size))
-
+                new_candidates = self.structure_scan(val_struct_provider, depth, recursive_limit_size)
+                candidates.extend(new_candidates)
```

Comment:
Is there any reason to add `new_candidates` and extend it to `candidates`?

Reply:
Only easiest debug purpose in single file.

Reply:
I see.